Kmeans_python/fit.py

```diff
@@ -30,5 +106,64 @@ def fit(X_train, k):
     >>> centers, labels = fit(X, 2)
     
     """
+    if not ((isinstance(X_train, pd.DataFrame)) | (isinstance(X_train, np.ndarray))):
+        raise ValueError("Invalid input type for samples."
+                         " X_train must be pandas dataframe or a numpy array.")
+        
+    if n_init <= 0:
+        raise ValueError("Invalid number of initializations."
+                         " n_init=%d must be bigger than zero." % n_init)
+
+    if max_iter <= 0:
+        raise ValueError("Number of interations should be a positive integer."
+                         " max_iter=%d must be bigger than zero." % max_iter)
+    
+    if k < 1:
+        raise ValueError("Invalid number of clusters."
+                         " k=%d must be an integer bigger than or equal to 1." % k)
+    
+    
+    if isinstance(X_train, pd.DataFrame):
+        
+        X = X_train.to_numpy()
+    else:
+        X = X_train
+        
+    n_samples, n_features = X.shape
+    
+    labels = np.empty((n_samples, 1))
+        
+    inertia = np.inf
+    
+    i = 0
+    
+    while i <= n_init:
+        
+        # randomly initializing centers
+        centers = random_init(X, k, n_features)
+        
+        for j in range(max_iter):
+            
+            # calculating euclidean distance of each sample from the centers
+            centroid_distances = compute_distance(X, centers)
+
+            labels = np.argmin(centroid_distances, axis = 1)
+            
+            # updating the centers with mean
+            for m in range(k):
+                
+                centers[m,:] = np.mean(X[np.where(labels == m)[0], : ], axis = 0)
+        
+        updated_inertia = np.sum(np.square(centroid_distances))
+
```

Comment:
I believe, inertia is not the sum of the squared distance of all samples to all cluster centers. But the sum of the squared distance of all samples to their respective cluster centers (sum of intra-cluster squared distance)

```suggestion
            inertia_array = np.zeros((k, 1))
            for m in range(k):                
                centers[m,:] = np.mean(X[np.where(labels == m)[0], : ], axis = 0)
                distance_m = np.min(centroid_distances[np.where(labels == m)[0], : ], axis = 0)
                inertia_array[m, 0] = np.sum(np.square(distance_m))
        
        # Add all intra-cluster distances to get the inertia
        updated_inertia = np.sum(inertia_array)

```

Reply:
I will look into this and update the code